lametro/api.py

```diff
@@ -110,24 +110,45 @@ def get_relations(self, concepts):
                 for term in concepts["terms"]
             }
 
-        return self.filter_concepts(relations)
+        return self.map_to_subject(relations)
 
-    def filter_concepts(self, concepts):
+    def map_to_subject(self, concepts, filter_concepts=True):
         result_count = int(self.request.GET.get("maxResultCount", 10))
-        guids = list(concepts.keys())
-        subjects = list(
-            LAMetroSubject.objects.filter(guid__in=guids).values("name", "guid")
-        )[:result_count]
-
-        for subject in subjects:
-            subject["display_name"] = subject["name"] + concepts[subject["guid"]].get(
-                "synonym_label", ""
+
+        matched_subjects = {
+            sub["guid"]: {"name": sub["name"]}
+            for sub in LAMetroSubject.objects.filter(
+                guid__in=list(concepts.keys())
+            ).values("name", "guid")
+        }
+
+        subjects = []
+
+        for guid, concept in concepts.items():
+            # If only returning concepts that we can map to a subject,
+            # continue if there is not a corresponding subject
+            if filter_concepts:
+                base_obj = matched_subjects.get(guid)
+                if not base_obj:
+                    continue
+
+            # Otherwise, fall back to the concept as it is represented
+            # in the SES API
+            else:
+                base_obj = matched_subjects.get(guid, concept)
+
+            subjects.append(
+                {
+                    "name": base_obj["name"],
+                    "display_name": base_obj["name"] + concept.get("synonym_label", ""),
+                    "guid": guid,
+                }
             )
 
         return {
             "status_code": 200,
             "concepts": concepts,
-            "subjects": subjects,
+            "subjects": subjects[:result_count],
```

Comment:
What is this bracket/colon combo doing? It was a bit hard looking this one up.

Reply:
This is called [slice notation](https://realpython.com/lessons/indexing-and-slicing/)! Here, it returns up to the requested number of results from `subjects`.

Reply:
Ooooh gotcha, ty! For some reason I had thought that `subjects` here was a dict, not a list. This makes a lot of sense.